crates/daphne/src/protocol/mod.rs

```diff
@@ -595,6 +713,51 @@ mod test {
 
     test_versions! { agg_job_resp_abort_invalid_transition }
 
+    fn finish_agg_job_vdaf_draft09(version: DapVersion) {
+        let t = AggregationJobTest::new(TEST_VDAF_DRAFT09, HpkeKemId::X25519HkdfSha256, version);
+        let reports = t.produce_reports(vec![
+            DapMeasurement::U64(1),
+            DapMeasurement::U64(1),
+            DapMeasurement::U64(0),
+            DapMeasurement::U64(0),
+            DapMeasurement::U64(1),
+        ]);
+
+        let (leader_state, agg_job_init_req) =
+            t.produce_agg_job_req(&DapAggregationParam::Empty, reports);
+
+        let (leader_agg_span, helper_agg_span) = {
+            let (helper_agg_span, agg_job_resp) = t.handle_agg_job_req(agg_job_init_req);
+            let leader_agg_span = t.consume_agg_job_resp(leader_state, agg_job_resp);
+
+            (leader_agg_span, helper_agg_span)
+        };
+
+        assert_eq!(leader_agg_span.report_count(), 5);
+        let num_measurements = leader_agg_span.report_count();
+
+        let VdafAggregateShare::Field64Draft09(leader_agg_share) =
+            leader_agg_span.collapsed().data.unwrap()
+        else {
+            panic!("unexpected VdafAggregateShare variant")
+        };
+
+        let VdafAggregateShare::Field64Draft09(helper_agg_share) =
+            helper_agg_span.collapsed().data.unwrap()
+        else {
+            panic!("unexpected VdafAggregateShare variant")
+        };
+
+        let vdaf = Prio3Draft09::new_count(2).unwrap();
+        assert_eq!(
+            vdaf.unshard(&(), [leader_agg_share, helper_agg_share], num_measurements,)
+                .unwrap(),
+            3,
+        );
+    }
+
+    test_versions! { finish_agg_job_vdaf_draft09 }
```

Comment:
This macro creates two tests, one for draft09 and one for the latest draft. Let's just make `finish_agg_job_vdaf_draft09` a test.